packages/fx-core/templates/plugins/resource/aad/auth/V3/README.txt

```diff
@@ -1 +1,663 @@
+                OAuthAuthority: ${{AAD_APP_OAUTH_AUTHORITY}}
+      ```
+  -------------------------
+
+4. Update Infra
+  AAD related configs needs to be configure to your remote service. Following example shows the configs on Azure Webapp.
+    a. TeamsFx__Authentication__ClientId: AAD app client id
+    b. TeamsFx__Authentication__ClientSecret: AAD app client secret
+    c. TeamsFx__Authentication__OAuthAuthority: AAD app oauth authority
+  
+  -------------------------
+  Example for TeamsFx Tab template
+  -------------------------
+  Open `infra/azure.parameter.json` and add following lines into `parameters`:
+  ```
+  "tabAadAppClientId": {
+    "value": "${{AAD_APP_CLIENT_ID}}"
+  },
+  "tabAadAppClientSecret": {
+    "value": "${{SECRET_AAD_APP_CLIENT_SECRET}}"
+  },
+  "tabAadAppOauthAuthorityHost": {
+    "value": "${{AAD_APP_OAUTH_AUTHORITY_HOST}}"
+  },
+  "tabAadAppTenantId": {
+    "value": "${{AAD_APP_TENANT_ID}}"
+  }
+  ```
+
+  Open `infra/azure.bicep` find follow line:
+  ```
+  param location string = resourceGroup().location
+  ```
+  and add following lines:
+  ```
+  param tabAadAppClientId string
+  param tabAadAppOauthAuthorityHost string
+  param tabAadAppTenantId string
+  @secure()
+  param tabAadAppClientSecret string
+  ```
+  and add following configs in webApp.properties.siteConfig.appSettings
+  ```
+  {
+    name: 'TeamsFx__Authentication__ClientId'
+    value: tabAadAppClientId
+  }
+  {
+    name: 'TeamsFx__Authentication__ClientSecret'
+    value: tabAadAppClientSecret
+  }
+  {
+    name: 'TeamsFx__Authentication__OAuthAuthority'
+    value: uri(tabAadAppOauthAuthorityHost, tabAadAppTenantId)
+  }
+  ```
+  -------------------------
+
+5. Update `appsettings.json` and `appsettings.Development.json`
+  AAD related configs needs to be configure to your .Net project settings:
+    ```
+    TeamsFx: {
+      Authentication: {
+        ClientId: AAD app client id
+        ClientSecret: AAD app client secret,
+        OAuthAuthority: AAD app oauth authority
+      }
+    }
+    ```
+  Note: You can use use $ENV_NAME$ to reference envs in local/remote service.
+
+  -------------------------
+  Example for TeamsFx Tab template
+  -------------------------
+  Open `appsettings.json` and `appsettings.Development.json`, and append the following lines:
+  ```
+  "TeamsFx": {	
+    "Authentication": {	
+      "ClientId": "$clientId$",	
+      "ClientSecret": "$client-secret$",	
+      "OAuthAuthority": "$oauthAuthority$"	
+    }	
+  }
+  ```
+  -------------------------
+
+6. Update source code
+  With all changes above, your env is ready and can update your code to add SSO to your Teams app.
+  You can find samples in following pages:
+    - TeamsFx SDK: https://www.nuget.org/packages/Microsoft.TeamsFx/
+    - Sample Code: under `TeamsFx-Auth/Tab`
+  
+  -------------------------
+  Example for TeamsFx Tab template
+  -------------------------
+  1) Create `Config.cs` and paste the following code:
```

Comment:
where do we consume config.cs? Seems our sample code does not use it

Reply:
Add changes in program.cs where consume this file. Leave a todo to see whether we can improve non sso tab template